lib/streamlit/app_session.py

```diff
@@ -167,8 +175,10 @@ def shutdown(self):
                 self._enqueue_script_request(ScriptRequest.SHUTDOWN)
 
             self._state = AppSessionState.SHUTDOWN_REQUESTED
-            self._local_sources_watcher.close()
-            self._stop_config_listener()
+            if self._local_sources_watcher:
+                self._local_sources_watcher.close()
+            if self._stop_config_listener is not None:
+                self._stop_config_listener()
```

Comment:
Why `is not None` for one of these if clauses but not the other? (Since Python has truthy/falsy typing, I'm almost always in favor of the verbose, explicit `is not None`, FWIW. We seem to only enforce this on the frontend but I'd love to enforce it everywhere.)

Reply:
Also, when will _local_sources_watcher ever be None? It looks like it's non-optional, so it should always be defined, right?